usecases/objects/auto_schema.go

```diff
@@ -83,11 +83,16 @@ func (m *autoSchemaManager) autoSchema(ctx context.Context, principal *models.Pr
 			return err
 		}
 		if schemaClass == nil {
-			return m.createClass(ctx, principal, object.Class, properties)
+			if err := m.createClass(ctx, principal, object.Class, properties); err != nil {
+				return err
+			}
+			continue
 		}
-
-		if _, err := m.schemaManager.AddClassProperty(ctx, principal, schemaClass, true, properties...); err != nil {
-			return err
+		newProps := schema.DedupProperties(schemaClass.Properties, properties)
```

Comment:
**question:** is it needed ? given that it's called again inside `AddClassProperty()` https://github.com/weaviate/weaviate/pull/4680/files#diff-5544b193719c3fc2fd042bea272df1170c1512a3f9c103d112e0208504362fbdR71

Reply:
That is correct, the same method is called there again. DedupProperties called here for 2 reasons:
- cached class should be invalidated only when class is changed. If props are passed to AddClassProperty and all of them already exist in schema, cache would be invalidated unnecessarily (used in incoming autoschema PR)
- existing properties does not have to be validated / extended with default values / migrated. That way only unknown to class schema properties are passed for further processing.